File: 1). DSA/1). Data Structures/01). Array/C++/_003)_Delete_Element.cpp
#include <iostream>
#include <cmath>
using namespace std;

int deleteEle(int arr[], int n, int x)
{
	int i = 0;
	for(i = 0; i < n; i++)
	{
		if(arr[i] == x)
			break;
	}
	if(i == n)
		return n;
	for(int j = i; j < n - 1; j++)
	{
		arr[j] = arr[j + 1];
	}
	return n-1;
}

int main()
{
       int arr[] = {3, 8, 12, 5, 6}, x = 12, n = 5;
       cout<<"Before Deletion"<<endl;
       for(int i=0; i < n; i++)
       {
       	cout<<arr[i]<<" ";
       }
       cout<<endl;
       n = deleteEle(arr, n, x);
       cout<<"After Deletion"<<endl;
       for(int i=0; i < n; i++)
       {
       		cout<<arr[i]<<" ";
       }
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_003)_Delete_Element.dart
List<int> delete(arr, element) {
  if (arr.indexOf(element) == -1) {
    throw new Exception("Element does not exist in array");
  }
  var newArr = <int>[];
  for (int i = 0; i < arr.length; i++) {
    if (arr[i] != element) {
      newArr.add(arr[i]);
    }
  }
  return newArr;
}

void main() {
  var arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
  print('Before Deletion: ${arr}');
  var delArr = delete(arr, 6);
  print('After Deletion: ${delArr}');
}
